From bdfd1236ee01c7595a093158f88089159c313cd3 Mon Sep 17 00:00:00 2001
From: Aaron Ruby <aruby@blackberry.com>
Date: Thu, 13 Jul 2023 18:29:11 -0400
Subject: [PATCH 1/2] Modifications for QNX

---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index c14b7f5f1..05773b48f 100644
--- a/meson.build
+++ b/meson.build
@@ -552,6 +552,8 @@ if cc.get_id() == 'gcc' or cc.get_id() == 'clang'
     '-Werror=missing-include-dirs',
     '-Werror=pointer-arith',
     '-Werror=unused-result',
+    '-Wno-nonnull',
+    '-Wno-format-nonliteral',
   ]
 
   warning_c_args = warning_common_args + [
-- 
2.34.1

