From 8dc226934ce208c6a6e342907ca5275654db053b Mon Sep 17 00:00:00 2001
From: Aaron Ruby <aruby@blackberry.com>
Date: Thu, 13 Jul 2023 18:22:47 -0400
Subject: [PATCH] Modifications for QNX

---
 meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/meson.build b/meson.build
index 6fe47a29d..cbac3117a 100644
--- a/meson.build
+++ b/meson.build
@@ -66,6 +66,9 @@ gmodule_dep = dependency('gmodule-no-export-2.0', version: glib_req_version,
                          fallback : ['glib', 'libgmodule_dep'])
 gio_dep = dependency('gio-2.0', version: glib_req_version,
                      fallback : ['glib', 'libgio_dep'])
+## BEGIN QNX changes
+libsocket_dep = dependency('libsocket')
+## END QNX changes
 
 # Configurations
 gdk_pixbuf_conf = configuration_data()
@@ -227,6 +230,9 @@ gdk_pixbuf_deps = [
   shared_mime_dep,
   medialib_dep,
   intl_dep,
+## BEGIN QNX changes
+  libsocket_dep,
+## END QNX changes
 ]
 
 # Check if we can build shared modules
-- 
2.34.1

